### ISSUE TITLE
CA1065: Incorrect detection inside indexer

### ISSUE THREAD

**Issue by skarllot** [dotnet/roslyn-analyzers#1842]
#### Analyzer package

[Microsoft.CodeQuality.Analyzers](https://www.nuget.org/packages/Microsoft.CodeQuality.Analyzers)

#### Package Version

v2.6.2

#### Diagnostic ID

[CA1065](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1065-do-not-raise-exceptions-in-unexpected-locations?view=vs-2017)

#### Repro steps

Create a class with the following property:
```csharp
public T2 this[T1 value1]
{
    get
    {
        int i = FindEntryByValue1(value1);
        if (i < 0)
            throw new KeyNotFoundException("The given value was not present in the map.");

        return entries[i].Value2;
    }

    set
    {
        bool modified = TryInsert(value1, value, InsertionBehavior.OverwriteExistingValue2);
        if (!modified)
            throw new ArgumentException("The given value is already present in the map.");
    }
}
```

#### Expected behavior

Should ignore the `KeyNotFoundException`, because it is accepted inside indexed gets.

#### Actual behavior

An `CA1065` issue is incorrectly detected.
